packages/backend/src/test/unit/metric.service.spec.ts

```diff
@@ -0,0 +1,198 @@
+        expect(spyAggregateRequestMetric.mock.calls[2][0]).toBe(records[2]);
+      });
+
+      it('fetchMetrics should call aggregateNetworkMetric', async () => {
+        const spyGetRecords = jest.spyOn((metricService as any), 'getRecords').mockResolvedValue(records);
+        const spyNetworkMetric = jest.spyOn((metricService as any), 'aggregateNetworkMetric').mockImplementation();
+
+        await (metricService as any).fetchMetrics(MetricsChannels.NETWORK, chunk);
+
+        expect(spyGetRecords).toBeCalledTimes(1);
+        expect(spyGetRecords).toBeCalledWith(MetricsChannels.NETWORK, chunk);
+        expect(spyNetworkMetric).toBeCalledTimes(records.length);
+        expect(spyNetworkMetric.mock.calls[0][0]).toBe(records[0]);
+        expect(spyNetworkMetric.mock.calls[1][0]).toBe(records[1]);
+        expect(spyNetworkMetric.mock.calls[2][0]).toBe(records[2]);
+      });
+    });
+
+    it('aggregateNetworkMetric should create network-metric entity', async () => {
+      const data = {
+        network: {
+          bytesIn: 1,
+          bytesOut: 2,
+          connections: 3,
+        },
+        date: 4,
+        nodeId: 'nodeId',
+      }
+      const spyCreate = jest.spyOn((metricService as any).networkMetricModel, 'create').mockImplementation();
+
+      await (metricService as any).aggregateNetworkMetric(JSON.stringify(data));
+
+      expect(spyCreate).toBeCalledTimes(1);
+      expect(spyCreate).toBeCalledWith({ nodeId: data.nodeId, date: data.date, ...data.network });
+    });
+
+    it('aggregateRequestMetric should create request-metric entity', async () => {
+      const nodeId = '1.2.3';
+      const userAgent = 'userAgent';
+      const ip = '1.2.3';
+      const query = { query: 'query {a {a}}', variables: {} };
+      const request = { smth: 'smth' };
+      const rawResponseBody = 'rawResponseBody';
+      const contentLength = 'contentLength';
+      const requestDate = 1;
+      const responseDate = 2;
+
+      const records: AnyMetric[] = [
+        {
+          event: MetricsChannels.GOT_REQUEST,
+          nodeId,
+          query,
+          userAgent,
+          ip,
+          request,
+          date: requestDate,
+        } as any,
+        {
+          event: MetricsChannels.SENT_RESPONSE,
+          rawResponseBody,
+          contentLength,
+          date: responseDate,
+        },
+      ]
+      const resolvers = [{ path: 'path', latency: 100500 }];
+
+      const spyCreate = jest.spyOn((metricService as any).requestMetricModel, 'create').mockImplementation();
+      const spyLrange = jest.spyOn((metricService as any).redis, 'lrange').mockResolvedValue(records.map(r => JSON.stringify(r)));
+      const spyLtrim = jest.spyOn((metricService as any).redis, 'ltrim').mockImplementation(() => {});
+      const spyReduceResolvers = jest.spyOn((metricService as any), 'reduceResolvers').mockReturnValue(resolvers);
+
+      const requestId = 1;
+      await (metricService as any).aggregateRequestMetric(requestId);
```

Comment:
type mismatch warning

Reply:
aggregateRequestMetric it's private method, is there another way for testing besides "any"?

Reply:
it's actually about the argument of the function which accepts string, but `requestId` is number.